### PR TITLE
Geodesic methods

### LINKED ISSUE #380: Add __repr__ and __str__ for Geodesic Class
🐞 **Problem**

The Geodesic class is missing these methods. More can be discusses in comments.

🎯 **Goal**

💡 **Possible solutions**

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by Bhavam**
Added `__str__` and `__repr__` methods to `Geodesic` class

Fixes #380 

@shreyasbapat

### CODE REVIEW

**Comment by shreyasbapat** [comment]
Hi @Bhavam You need to fix the code quality.

Run `tox -e reformat` 

while in the repo.

**Comment by shreyasbapat** [comment]
Also, can you define a example geodesic and show how the representation looks like? Pick anyone from the example notebook

**Comment by Bhavam** [comment]
Yes sure ill do that and get back to you.

File: src/einsteinpy/geodesic.py [hunk 1]
```diff
@@ -32,6 +32,12 @@ def __init__(
             return_cartesian=True,
         )[1]
 
+    def __repr__(self):
+        return {'body' : self.body, 'attractor' : self.attractor , 'metric': self.metric , '_trajectory': self._trajectory}
+
+    def __str__(self):
+        return 'Geodesic(body=' + self.body , 'Geodesic(attractor=' + self.attractor , 'Geodesic(metric' + self.metric , 'Geodesic(_trajectory=' + self._trajectory
+
```
Comment (ritzvik) [inline src/einsteinpy/geodesic.py]: It is always better to insert variables into string like this.
```
"something {} again something {}".format(obj1, obj2)
```

or

```
"".join("something", obj1, "again something", obj2)
```

File: src/einsteinpy/geodesic.py [hunk 2]
```diff
@@ -32,6 +32,12 @@ def __init__(
             return_cartesian=True,
         )[1]
 
+    def __repr__(self):
+        return {'body' : self.body, 'attractor' : self.attractor , 'metric': self.metric , '_trajectory': self._trajectory}
+
```
Comment (ritzvik) [inline src/einsteinpy/geodesic.py]: `__repr__` should return a formal string representation of the object. I guess, you are returning a `dict`

**Comment by shreyasbapat** [comment]
Good. Now I believe, the representations won't pop up. So, please try to run it once.

File: src/einsteinpy/geodesic.py (hunk 2 above)
Comment (shreyasbapat) [reply src/einsteinpy/geodesic.py]: I guess teh trajectory must be skipped, it is an array. Just the details about the body, its parent body, its parent body's mass, and what metric it uses. That would work.

**Comment by shreyasbapat** [comment]
And later you will have to write up a unit test that tests this change to take care of the failing coverage check.

**Comment by Bhavam** [comment]
Sure and ill follow up with the example geodesic representation also.

File: src/einsteinpy/geodesic.py (hunk 2 above)
Comment (shreyasbapat) [reply src/einsteinpy/geodesic.py]: I guess you also would like to add `self.body.parent.name` and `self.body.parent.mass`

Comment (Bhavam) [reply src/einsteinpy/geodesic.py]: Yes on it , I assume this way of creating the `__str__` and `__repr__` methods is agreeable then?

**Comment by Bhavam** [comment]
> And later you will have to write up a unit test that tests this change to take care of the failing coverage check.

Do I insert the unit tests in the testing module or somewhere else

**Comment by ritzvik** [comment]
> > And later you will have to write up a unit test that tests this change to take care of the failing coverage check.
> 
> Do I insert the unit tests in the testing module or somewhere else

Kindly put in new tests in `tests/test_geodesic.py`
Also, run `tox -e reformat` to lint the code and pass quality checks :)

File: src/einsteinpy/geodesic.py [hunk 3]
```diff
@@ -32,6 +32,24 @@ def __init__(
...
+            self.body.parent.name,
+            self.body.parent.mass,
+        )
+
+    def __str__(self):
+        return "body name= ({0}) , attractor name=({1}) , metric=({2}) , parent name=({3}) , parent mass=({4})".format(
+            self.body.name,
+            self.attractor.name,
+            self.metric.name,
+            self.body.parent.name,
```
Comment (shreyasbapat) [inline src/einsteinpy/geodesic.py]: I guess this is same as attractor

File: src/einsteinpy/metric/schwarzschild.py [hunk 4]
```diff
@@ -19,6 +19,7 @@ class Schwarzschild:
 
     @u.quantity_input(time=u.s, M=u.kg)
     def __init__(self, sph_coords, M, time):
+        self.name = "Schwarzschild"
```
Comment (shreyasbapat) [inline src/einsteinpy/metric/schwarzschild.py]: You need to add this in other two metric as well, Kerr and KerrNewman

**Comment by shreyasbapat** [comment]
I guess all you have to do is,
Squash this in one commit.

And rebase.

If you need any help in that front, ping me and I'll be happy to help.

**Comment by Bhavam** [comment]
Squashed and rebased , thanks and let me know if any more changes are required from my side.

**Comment by shreyasbapat** [comment]
So, I guess. Congratulations on your first contribution :D 

I look forward to your contributions in coming days :)

**Comment by Bhavam** [comment]
Thanks for all your help and surely expect more contributions from my side,
Cheers.

**Pull request merged**